be/src/io/fs/file_system.h

```diff
@@ -64,6 +65,43 @@ struct FileInfo {
     bool is_file;
 };
 
+struct FsListGenerator {
+    FsListGenerator() = default;
+    virtual ~FsListGenerator() = default;
+
+    virtual Status init() = 0;
+
+    Result<FileInfo> next() {
```

Comment:
just make it pure virtual and eliminate `generageNext()`,
calling it here is not a good design.